src/main/scala/com/lightbend/rp/sbtreactiveapp/SbtReactiveAppPlugin.scala

```diff
@@ -20,6 +20,24 @@ import com.typesafe.sbt.packager.Keys._
 import com.typesafe.sbt.packager.archetypes.scripts.AshScriptPlugin
 import com.typesafe.sbt.packager.docker
 import sbt._
+import sbt.Keys._
+import sbt.plugins.JvmPlugin
+
+object SbtReactiveAppPluginAll extends AutoPlugin {
+  override def requires = JvmPlugin
+
+  override def trigger = allRequirements
+
+  override def projectSettings: Seq[Setting[_]] =
+    inConfig(docker.DockerPlugin.autoImport.Docker)(
+      publish in docker.DockerPlugin.autoImport.Docker := {
+        Def.taskDyn {
+          dockerRepository.?.value.nonEmpty && rpDockerPublish.?.value.nonEmpty
```

Comment:
Does this statement actually do anything? If this evaluates to `false`, `Def.task()` will still be returned?

Reply:
Yeah, but I agree it's confusing -- shame on me! The && is there to just short-circuit evaluation of `rpDockerPublish.?.value` if `dockerRepository.?.value` is `None`. `rpDockerPublish.?.value` will evaluate the `rpDockerPublish` task (which returns empty) if it's defined.

I've filed https://github.com/lightbend/sbt-reactive-app/issues/24 to fix this up.